x-pack/plugins/fleet/common/services/agent_status_filters.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AGENT_POLLING_THRESHOLD_MS } from '../constants';
+
+function buildKueryForEnrollingAgents(path: string = ''): string {
+  return `not (${path}last_checkin:*)`;
+}
+
+function buildKueryForUnenrollingAgents(path: string = ''): string {
+  return `${path}unenrollment_started_at:*`;
+}
+
+function buildKueryForUnenrolledAgents(path: string = ''): string {
+  return `${path}unenrolled_at:*`; // active:false
```

Comment:
@hop-dev should unenrolled be `active:false`? it seems to be that way in the new logic

```
    if (${field('active')}.size() > 0 && ${field('active')}.value == false) {
      emit('unenrolled'); 
```

Reply:
Yes I believe active false is correct, unenrolled_at should also work but I think active felt a more reliable field